File: app/src/main/java/br/com/nicetuber/feature/home/HomePresenter.java
package br.com.nicetuber.feature.home;

import java.util.List;

import javax.inject.Inject;

import br.com.nicetuber.base.BasePresenter;
import br.com.nicetuber.model.ChannelSearched;

/**
 * Created by thiagozg on 15/10/2017.
 */

public class HomePresenter extends BasePresenter<IHome.View, HomeInteractor>
    implements IHome.Callback {

    @Inject
    public HomePresenter() {
    }

    public void searchChannel(String query) {
        interactor.searchChannel(this, query);
    }

    @Override
    public void onSearchChannelSuccess(List<ChannelSearched> response) {
        view.showList(response);
    }

    @Override
    public void onSearchChannelError(String message) {
        view.showErrorMessage(message);
    }
}

File: app/src/main/java/br/com/nicetuber/feature/home/HomePresenter.kt
package br.com.nicetuber.feature.home

import br.com.nicetuber.base.BasePresenter
import br.com.nicetuber.model.ChannelSearched
import javax.inject.Inject

/**
 * Created by thiagozg on 15/10/2017.
 */

class HomePresenter @Inject
constructor() : BasePresenter<IHome.View, HomeInteractor>(), IHome.Callback {

    fun searchChannel(query: String) {
        interactor.searchChannel(this, query)
    }

    override fun onSearchChannelSuccess(response: List<ChannelSearched>) {
        view?.showList(response)
    }

    override fun onSearchChannelError(message: String) {
        view?.showErrorMessage(message)
    }
}
